extensions/wikibase/src/org/openrefine/wikibase/qa/scrutinizers/FileNameScrutinizer.java

```diff
@@ -78,15 +78,20 @@ public void scrutinize(MediaInfoEdit edit) {
                 seenFileNames.add(normalizedFileName);
             }
 
-            // check whether filenames exceed the maximum length (240 bytes, which we take to be 240 characters)
-            // TODO check for the length as bytes in the corresponding enconding (which one?)
+            // check whether filenames exceed the maximum length (240 bytes, coded as UTF-8)
             // see https://commons.wikimedia.org/wiki/Commons:File_naming
-            if (fileName.length() > maxFileNameLength) {
-                QAWarning issue = new QAWarning(fileNameTooLongType, null, QAWarning.Severity.CRITICAL,
-                        1);
-                issue.setProperty("example_filename", fileName);
-                issue.setProperty("max_length", Integer.toString(maxFileNameLength));
-                addIssue(issue);
+            try {
+                byte[] utf8Bytes = fileName.getBytes("UTF-8");
+                if (utf8Bytes.length > maxFileNameBytes) {
+                    QAWarning issue = new QAWarning(fileNameTooLongType, null,
+                            QAWarning.Severity.CRITICAL, 1);
+                    issue.setProperty("example_filename", fileName);
+                    // This is only accurate for 1 byte characters
+                    issue.setProperty("max_length", Integer.toString(maxFileNameBytes));
+                    addIssue(issue);
+                }
+            } catch (UnsupportedEncodingException e) {
+                e.printStackTrace();
```

Comment:
I'd remove this `e.printStackTrace()` because it would be printed once for each row every time the issues are refreshed. I think it's safe to just ignore this exception altogether because if UTF-8 is not supported on the system we have bigger problems than not being able to report a quality assurance warning.